src/utilities/formatter.js

```diff
@@ -0,0 +1,46 @@
+const prettier = require('prettier');
+const stripAnsi = require('strip-ansi');
+
+function formatter(unformattedCode, language) {
+  let options = {
+    trailingComma: 'none',
+  };
+
+  switch (language) {
+    case 'html':
+      options.parser = 'html';
+      break;
+
+    case 'css':
+      options.parser = 'css';
+      break;
+    case 'js':
+      options.parser = 'babel';
+      break;
+    case 'json':
+      options.parser = 'json';
+      break;
+    case 'markdown':
+      options.parser = 'markdown';
+      break;
+    case 'yaml':
+      options.parser = 'yaml';
+      break;
+  }
+
+  console.log(options);
+
+  let formattedCode = '';
+
+  try {
+    formattedCode = prettier.format(unformattedCode, options);
+  } catch (error) {
+    formattedCode = stripAnsi(
+      `SyntaxError: Unexpected token (${error.loc.start.line}:${error.loc.start.column})\n\n${error.codeFrame}`
+    );
```

Comment:
What does the stripAnsi do and why is it needed? This looks like magic to me atm.

Reply:
When formatting code if Prettier's parser detects a syntax error, it stops parsing and shows an error by calling `console.error`. But I catch that error and send it back to Discord as message from the bot. But the problem is that Prettier _assumes_ that their error message will be displayed only in the console, so to help the devs figure out where a syntax error occurred, they colorize their error message with ANSI. This isn't a problem when it comes to console (who doesn't like colorful errors?) but if I send it to Discord, it can't display it. That's why we have to use `stripAnsi` to remove ANSI from the error message.

Shout out to @Ryuno-Ki as he helped me solve this problem as he previously ran into it where as I had no clue what went wrong!

Reply:
Correct. We want to defer to Discord for syntax highlighting. Prettier is applying some ANSI styles as well (via Chalk).
I suggested three options:
- using a library, like stripAnsi (c.f. https://github.com/prettier/prettier/issues/2956#issuecomment-333597851 )
- use `--no-color` CLI argument (c.f. https://github.com/prettier/prettier/issues/4118#issue-303283804 )
- set `FORCE_COLOR=0` environment variable (c.f. https://github.com/prettier/prettier/issues/4118#issuecomment-371401955 )

Reply:
Done!